record_serializer_test.go

```diff
@@ -358,3 +359,68 @@ func TestTemplateRecordSerializer(t *testing.T) {
 		})
 	}
 }
+
+func TestTemplateSerializer_MissingKey(t *testing.T) {
+	tests := []struct {
+		name        string
+		template    string
+		record      opencdc.Record
+		expectError bool
+	}{
+		{
+			name:     "valid template with existing key",
+			template: "{{.Key}}",
+			record: opencdc.Record{
+				Key: opencdc.RawData("test-key"),
+			},
+			expectError: false,
+		},
+		{
+			name:     "template with missing key",
+			template: "{{.NonExistentKey}}",
+			record: opencdc.Record{
+				Key: opencdc.RawData("test-key"),
+			},
+			expectError: true,
+		},
+		{
+			name:     "nested template with missing key",
+			template: "{{.Metadata.nonexistent}}",
+			record: opencdc.Record{
+				Metadata: opencdc.Metadata{
+					"existing": "value",
+				},
+			},
+			expectError: true,
+		},
+		{
+			name:     "multiple expressions with missing key",
+			template: "{{.Key}} {{.NonExistentKey}}",
+			record: opencdc.Record{
+				Key: opencdc.RawData("test-key"),
+			},
+			expectError: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			serializer := TemplateRecordSerializer{}
```

Comment:
The assertions in this test could be simplified if we used `is`, like we do in most of our tests. Here's a [random example](https://github.com/ConduitIO/conduit-connector-sdk/blob/ca006803b8f497821949c0801660fd153cd46eb6/destination_middleware_test.go#L140-L147).